front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/documents/index.ts

```diff
@@ -19,33 +19,38 @@ async function handler(
   res: NextApiResponse<WithAPIErrorResponse<GetDocumentsResponseBody>>
 ): Promise<void> {
   const session = await getSession(req, res);
+  const auth = await Authenticator.fromSuperUserSession(
+    session,
+    req.query.wId as string
+  );
 
-  const { wId, name } = req.query;
-  if (typeof wId !== "string" || typeof name !== "string") {
+  if (!auth.isDustSuperUser()) {
     return apiError(req, res, {
-      status_code: 400,
+      status_code: 404,
       api_error: {
-        type: "invalid_request_error",
-        message: "The request query parameters are invalid.",
+        type: "user_not_found",
+        message: "Could not find the user.",
       },
     });
   }
 
-  const auth = await Authenticator.fromSuperUserSession(session, wId);
-  const owner = auth.workspace();
-
-  if (!owner || !auth.isDustSuperUser()) {
+  const { dsId } = req.query;
+  if (typeof dsId !== "string") {
     return apiError(req, res, {
       status_code: 404,
       api_error: {
-        type: "user_not_found",
-        message: "Could not find the user.",
+        type: "invalid_request_error",
+        message: "Invalid path parameters.",
       },
     });
   }
 
-  const dataSource = await getDataSource(auth, name);
-
+  const dataSource = await DataSourceResource.fetchByNameOrId(
+    auth,
+    dsId,
+    // TODO(DATASOURCE_SID): Clean-up
```

Comment:
nit: what cleanup would be needed here? change the call to "fetchById" when we remove "name"? IMO we can remove the com (will be done auto by IDE renaming)